### PR TITLE
[SYCL][NFC] Move sampler_impl to source dir

### PR DESCRIPTION
**Pull request by romanovvlad**

### CODE REVIEW

**Comment by romanovvlad** [comment]
Need to rebase and update the test which check abi.

**Comment by romanovvlad** [comment]
> 
> 
> Need to rebase and update the test which check abi.

Done.

File: sycl/test/abi/symbol_size.cpp [hunk 1]
```diff
@@ -69,13 +68,6 @@ int main() {
 #ifdef __SYCL_DEVICE_ONLY__
   check_size<private_memory<int, 1>, 4>();
   check_size<detail::sampler_impl, 8>();
```
Comment (bader) [inline sycl/test/abi/symbol_size.cpp:70]: Should we bump library version with this change?

Comment (romanovvlad) [reply sycl/test/abi/symbol_size.cpp:70]: It's tricky question. But, I think, it's safer to bump. Will do.

Comment (romanovvlad) [reply sycl/test/abi/symbol_size.cpp:70]: Done.

**Pull request merged**